File: second_task/app/src/main/java/com/example/secondtask/main/first_fragment/InputFragment.java
package com.example.secondtask.main.first_fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.example.secondtask.R;
import com.example.secondtask.main.model.TetaMenza;
import com.example.secondtask.main.view_model.CustomViewModel;

public class InputFragment extends Fragment {

    private EditText first_name, last_name, age, OIB, workplace;
    private Button btn_save;
    private CustomViewModel viewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        viewModel = new ViewModelProvider(this.requireActivity()).get(CustomViewModel.class);

    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.input_fragment, container, false);

        first_name = rootView.findViewById(R.id.first_name);
        last_name = rootView.findViewById(R.id.last_name);
        age = rootView.findViewById(R.id.age);
        OIB = rootView.findViewById(R.id.OIB);
        workplace = rootView.findViewById(R.id.workplace);
        btn_save = rootView.findViewById(R.id.btn_save);

        setListeners();

        return rootView;
    }

    private void setListeners(){
        btn_save.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(first_name.getText().toString().equals("") || last_name.getText().toString().equals("") || age.getText().toString().equals("") ||
                        OIB.getText().toString().equals("") || workplace.getText().toString().equals("")){
                    Toast.makeText(getContext(),"Please fill all the fields before saving",Toast.LENGTH_LONG).show();
                    return;
                }
                viewModel.updateTete(new TetaMenza(first_name.getText().toString(),last_name.getText().toString(),
                                    workplace.getText().toString(),
                                    Integer.parseInt(age.getText().toString()),
                                    Integer.parseInt(OIB.getText().toString())));
                clearEditTextFields();
            }
        });
    }

    private void clearEditTextFields(){
        first_name.setText("");
        last_name.setText("");
        age.setText("");
        OIB.setText("");
        workplace.setText("");
    }
}

File: second_task/app/src/main/java/com/example/secondtask/main/first_fragment/InputFragment.kt
package com.example.secondtask.main.first_fragment

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import com.example.secondtask.R
import com.example.secondtask.databinding.InputFragmentBinding
import com.example.secondtask.main.model.TetaMenza
import com.example.secondtask.main.view_model.CustomViewModel

class InputFragment : Fragment() {

    private val viewModel : CustomViewModel by activityViewModels()
    private var _binding : InputFragmentBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = InputFragmentBinding.inflate(inflater,container,false)
        val view = binding.root

        setListeners()

        return view
    }

    fun setListeners(){
        binding.btnSave.setOnClickListener(){
            if(binding.firstName.text.toString().equals("") || binding.lastName.text.toString().equals("") || binding.OIB.text.toString().equals("") ||
                binding.age.text.toString().equals("") || binding.workplace.text.toString().equals("")){

                Toast.makeText(getContext(),"Please fill all the fields before saving",Toast.LENGTH_LONG).show();
            }
            else{
                viewModel.addTeta(TetaMenza(binding.firstName.text.toString(),binding.lastName.text.toString(),binding.workplace.text.toString(),
                                    binding.age.text.toString().toInt(),binding.OIB.text.toString().toInt() ))
                clearEditTextFields()
            }
        }
    }

    fun clearEditTextFields(){
        binding.firstName.text.clear()
        binding.lastName.text.clear()
        binding.OIB.text.clear()
        binding.workplace.text.clear()
        binding.age.text.clear()
    }
}